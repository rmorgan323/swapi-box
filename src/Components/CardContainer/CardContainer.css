.card-container-component {
  position: relative;
  z-index: 10;
  width: 1600px;
  margin: auto; }
  .card-container-component h2 {
    text-align: center;
    color: #fff; }

.card-container {
  display: flex;
  flex-wrap: wrap; }

@media screen and (max-width: 1599px) {
  .card-container-component {
    width: 1280px; } }

@media screen and (max-width: 1279px) {
  .card-container-component {
    width: 960px; } }

@media screen and (max-width: 959px) {
  .card-container-component {
    width: 640px; } }

@media screen and (max-width: 639px) {
  .card-container-component {
    width: 320px; } }
