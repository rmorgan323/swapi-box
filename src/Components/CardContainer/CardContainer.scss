.card-container-component {
	h2 {
		text-align: center;
		color: #fff;

	}
}

.card-container {
	display: flex;
	flex-wrap: wrap;
}