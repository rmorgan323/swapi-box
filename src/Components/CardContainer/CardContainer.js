import React from 'react';
import './CardContainer.css';
import Card from '../Card/Card.js';

const CardContainer = ({ containerTitle, favoriteCard, active, items }) => {

		let displayItems = items.filter(item => item.cardType === active)

		let currentDisplay = displayItems.map((item) => {
			return <Card name={item.name}
									 lineOne={item.lineOne}
									 lineTwo={item.lineTwo}
									 lineThree={item.lineThree}
									 lineFour={item.lineFour}
									 key={item.id} 
									 id={item.id}
									 cardType={item.cardType}
	                 favoriteCard={favoriteCard} />
		})
	
	return (
		<div className="card-container-component">
			<h2>{containerTitle}</h2>
			<div className="card-container">
				{currentDisplay}
			</div>
		</div>
	)
}

export default CardContainer;