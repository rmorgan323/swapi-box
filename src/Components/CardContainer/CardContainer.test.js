import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import CardContainer from './CardContainer';

describe.skip('CardContainer test', () => {
  it('should render correctly', () => {
    const renderedCardContainer = shallow(<CardContainer />);

    expect(renderedCardContainer.find('.CardContainer').length).toEqual(1);
  })
})