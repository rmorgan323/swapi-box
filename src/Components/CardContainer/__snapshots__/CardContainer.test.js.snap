// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    items={
        Array [
            Object {
              "cardType": 0,
              "favorite": true,
              "id": 1,
              "lineOne": "Tatooine",
              "lineThree": "200,000",
              "lineTwo": "Human",
              "name": "Luke Skywalker",
            },
            Object {
              "cardType": 1,
              "favorite": true,
              "id": 102,
              "lineFour": "Lobot",
              "lineOne": "gas giant",
              "lineThree": "temperate",
              "lineTwo": "6,000,000",
              "name": "Bespin",
            },
            Object {
              "cardType": 2,
              "favorite": false,
              "id": 207,
              "lineOne": "Digger Crawler",
              "lineThree": "30",
              "lineTwo": "wheeled",
              "name": "Sand Crawler",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-container"
>
        
</div>,
      "className": "card-container-component",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-container"
>
          
</div>,
        "className": "card-container-component",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
